src/hooks/useUniversalRouter.ts

```diff
@@ -56,7 +63,21 @@ export function useUniversalRouterSwapCallback(trade: InterfaceTrade | undefined
       }
       const gasLimit = calculateGasMargin(gasEstimate)
       response = await provider.getSigner().sendTransaction({ ...tx, gasLimit })
-    } catch (swapError: unknown) {
+    } catch (swapError) {
+      const reason = getReason(swapError)
+      if (
```

Comment:
Can this logic be moved to a helper function to improve readability/maintainability?